modules/mobile/app/controllers/mobile/v0/letters_controller.rb

```diff
@@ -109,13 +109,14 @@ def increment_download_counter
       # body params appear in the params hash in specs but not in actual requests
       def download_options_hash
         body_string = request.body.string
-        body_params = Rack::Utils.parse_nested_query(body_string)
-        permitted_params = body_params.keep_if { |k, _| k.in? DOWNLOAD_PARAMS }
-                                      .transform_values { |v| ActiveModel::Type::Boolean.new.cast(v) }
-                                      .transform_keys { |k| k.camelize(:lower) }
```

Comment:
Removed the key/value transformations because they're not really necessary. I copied them from the letters generator team's controller. The keys will already be camel case and the values are converted back to strings later.